src/components/ColorPicker.scss

```diff
@@ -6,186 +6,209 @@
   box-shadow: transparentize($oc-black, 0.75) 0px 1px 4px;
   border-radius: 4px;
   position: absolute;
+
   :root[dir="ltr"] & {
     left: -5.5px;
   }
+
   :root[dir="rtl"] & {
     right: -5.5px;
   }
-}
-
-.color-picker-control-container {
-  display: grid;
-  grid-template-columns: auto 1fr;
-  align-items: center;
-}
 
-.color-picker-triangle {
-  width: 0px;
-  height: 0px;
-  border-style: solid;
-  border-width: 0px 9px 10px;
-  border-color: transparent transparent $oc-white;
-  position: absolute;
-  top: -10px;
-  :root[dir="ltr"] & {
-    left: 12px;
-  }
-  :root[dir="rtl"] & {
-    right: 12px;
+  &-control-container {
```

Comment:
Not a fan of this. This will make it not searchable when you copy & search for a className.

Reply:
> The css was not scoped so scoped it for all files. This increases the readability and also makes sure that excalidraw css never gets applied globally which can affect when embedded inside some other component.

What you're doing doesn't increase any specificity/scope. Before, it was `.color-picker-control-container`, and after your change it still is `.color-picker-control-container`.

Reply:
> > The css was not scoped so scoped it for all files. This increases the readability and also makes sure that excalidraw css never gets applied globally which can affect when embedded inside some other component.
> 
> What you're doing doesn't increase any specificity/scope. Before, it was `.color-picker-control-container`, and after your change it still is `.color-picker-control-container`.
> 
By scoping I meant what I have done in  for example in `styles.scss` where there was no scoping and hence CSS was affecting the parent App so wrapped everything inside class `excalidraw` . And similarly wherever the scope was. missing I added it to make sure there is no global css

> If anything, this reduces readability because you then need to backtrack to the top of the file to figure out what the prefix is. And as I mentioned, searchability is gone.

This file didn't have any scope change, I just changed it shorthand sass syntax using `&` so that scss file doesn't look very cluttered. I understand your point of searching but do you feel that if it is one / two-level deeper it interferes with search? Since BEM is followed where class names gets bigger so I felt shorthand syntax will make it cleaner.

Reply:
> By scoping I meant what I have done in for example in styles.scss where there was no scoping and hence CSS was affecting the parent App so wrapped everything inside class excalidraw.

I missed that one. If that's the concern, shouldn't pretty much every class be wrapped in `excalidraw` namespace as it can potentially leak outside? That said, is this really a concern? You probably have a better idea than mine, but I thought if Excalidraw is used as a component in an app, it will live inside an `iframe`, so CSS namespacing should be an issue, no?

> This file didn't have any scope change, I just changed it shorthand sass syntax using & so that scss file doesn't look very cluttered. I understand your point of searching but do you feel that if it is one / two-level deeper it interferes with search?

It doesn't really matter how deeper it is, but if we define it using the shorthand, it's basically non-searchable at that point.

Reply:
> > By scoping I meant what I have done in for example in styles.scss where there was no scoping and hence CSS was affecting the parent App so wrapped everything inside class excalidraw.
> 
> I missed that one. If that's the concern, shouldn't pretty much every class be wrapped in `excalidraw` namespace as it can potentially leak outside? That said, is this really a concern? You probably have a better idea than mine, but I thought if Excalidraw is used as a component in an app, it will live inside an `iframe`, so CSS namespacing should be an issue, no?
> No When using npm package it won't be in an iframe. And as discussed earlier With npm package we have more control over the API and customizations/callbacks that is why I am working on creating the package so it can be easily integrated with any component hence we need to make sure we have the css scoped.
Since the outer css is already scoped inside excalidraw and in styles.css there is some css for example button which was affecting other files hence I had to overwrite using excalidraw namespace there as well.

And that being said since we are already scoping every scss file now so there is rarely any chance of getting it leaked unless the wrapper name is too generic which can easily collide with outer world.
And scoping is always better to have as it saves the App from outer world and also makes sure outer world is not affected due to the App.
Does that sound good ?

> > This file didn't have any scope change, I just changed it shorthand sass syntax using & so that scss file doesn't look very cluttered. I understand your point of searching but do you feel that if it is one / two-level deeper it interferes with search?
> 
> It doesn't really matter how deeper it is, but if we define it using the shorthand, it's basically non-searchable at that point.
> 
> We don't have any crazy deep BEM selectors right now, so I don't think we need to use a shorthand here.

Ok if we are not following that then I will revert those changes.